keras_cv/bounding_box/utils.py

```diff
@@ -87,3 +87,73 @@ def _format_outputs(boxes, squeeze):
     if squeeze:
         return tf.squeeze(boxes, axis=0)
     return boxes
+
+
+def pad_with_sentinels(bounding_boxes):
+    """Pads the given bounding boxes to convert it from a ragged tensor to
+    a regular tensor.
+
+    Args:
+      bounding_boxes: a ragged tensor of bounding boxes in 'corners' format.
+      Can be batched or unbatched.
+
+    Returns:
+      bounding_boxes_tensor: a tensor containing the -1 padded bounding boxes.
+    """
+    return bounding_boxes.to_tensor(-1)
+
+
+def filter_sentinels(bounding_boxes):
+    """filters the -1 padded bounding boxes to convert it from a regular
+       tensor to a ragged tensor.
+
+    Args:
+      bounding_boxes: a tensor of bounding boxes in 'corners' format.
+      Can be batched or unbatched.
+
+    Returns:
+      bounding_boxes_tensor: a ragged tensor containing filtered bounding boxes.
+    """
+    if isinstance(bounding_boxes, tf.Tensor):
+        tf.RaggedTensor.from_tensor(bounding_boxes)
+
+    def drop_padded_boxes(bounding_boxes):
+        bounding_boxes = bounding_boxes.to_tensor()
+        mask = bounding_boxes[:, 4] != -1
+        filtered_bounding_boxes = tf.boolean_mask(bounding_boxes, mask, axis=0)
+        return tf.RaggedTensor.from_tensor(filtered_bounding_boxes)
+
+    return tf.map_fn(drop_padded_boxes, bounding_boxes)
+
+
+def pad_with_class_id(bounding_boxes, class_id=0):
+    """pads bounding boxes with class id
+
+    Args:
+      bounding_boxes: a tensor of bounding boxes in 'corners' format.
+        Can be batched or unbatched.
+      class_id: The value of class id that needs to be padded.
+        The default value is 0
+
+    Returns:
+      bounding_boxes_tensor: a tensor containing class id padded bounding boxes.
+    """
+    if isinstance(bounding_boxes, tf.RaggedTensor):
```

Comment:
lets reformulate to not use if else, this can be handled by a single if:

```
is_ragged = isinstance(bounding_boxes, tf.RaggedTensor)

if is_ragged:
    row_lengths = ...
    bounding_boxes = bounding_boxes.to_tensor

... rest of the code.
```


As a general rule of thumb, if else statements are harder to read than if statements.

Reply:
noted.